### DISCUSSION TITLE
Auto confirm users via script

### DISCUSSION THREAD

**Discussion by etfz** [dani-garcia/vaultwarden#3954] [Q&A] [answered]
Hi,

I've been desperately looking for ways to simply the invitation process. I found this post a while ago, but never got around to testing it.
https://community.bitwarden.com/t/skip-user-confirmation/6969/6

Anyone got an idea of how one could integrate this smoothly into a Docker compose setup? Preferrably without building custom Vaultwarden images. I'm thinking something like having it run once a minute.

**Comment by BlackDex** [comment]
Have you checked https://bitwarden.com/help/cli/#serve and https://bitwarden.com/help/vault-management-api/
You could use an API key to login via the CLI and run a curl command to confirm them maybe?

**Comment by etfz** [comment]
I have. I know roughly which commands need to be used etc. I'm more looking to tie it all together, how to set it up in a Docker environment in a good way.

**Comment by etfz** [comment, marked as answer]
Revisited this. Realised Docker is irrelevant. Have not fully tested it yet, but you will want to download the [Bitwarden CLI client](https://bitwarden.com/help/cli/#download-and-install), configure it for your server using `bw config server https://vault.example.com`, and then set up a cronjob like so:

```
* * * * * bash bw-confirm-users.sh; bw logout
```

**bw-confirm-users.sh**
```bash
set -a
source .env.cli
set +a

bw login --raw --apikey

sessionKey="$(bw unlock --raw --passwordenv BW_PASSWORD)"

orgIds=$(bw --session $sessionKey list organizations | jq -r '.[].id')

for orgId in $orgIds; do
    members=$(bw --session $sessionKey list org-members --organizationid $orgId | jq -r '.[] | select(.status == 1) | .id')
    for memberId in $members; do
        echo "Confirming member $memberId..."
        bw confirm --session $sessionKey --organizationid $orgId org-member $memberId
    done
done
```

**.env.cli**
```
BW_CLIENTID=client_id
BW_CLIENTSECRET=client_secret
BW_PASSWORD=master_password
```
